### PR TITLE
Add an doc page for interoperatibility

### LINKED ISSUE #235: Document the `__cuda_stream__` protocol
One of the missing tasks for beta 1 (#79)... 😓

### PR DESCRIPTION
**Pull request by leofang**
Close #235.

### CODE REVIEW

File: cuda_core/docs/source/interoperability.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+we cover a list of possible such scenarios.
+
+
+Current device/context
+----------------------
+
+The :meth:`Device.set_current` method ensures that the calling host thread has
+an active CUDA context set to current. This CUDA context can be seen and accessed
+by other GPU libraries without any code change. For libraries built on top of
+the CUDA runtime (``cudart``), this is as if ``cudaSetDevice`` is called.
```
Comment (shwina) [inline cuda_core/docs/source/interoperability.rst]: `cudart` is an ambiguous term in this context, so we should clarify (could mean `cuda_bindings.cudart),

File: cuda_core/docs/source/interoperability.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+The :class:`~_stream.Stream` class is a vocabulary type representing CUDA streams
+in Python. While we encourage new Python projects to start using streams (and other
+CUDA types) from ``cuda.core``, we understand that there are already several projects
+exposing their own stream types.
+
+To address this issue, we propose the ``__cuda_stream__`` protocol (currently version
+0) as follows: For any Python objects that are meant to be interpreted as a stream, they
+should add a ``__cuda_stream__`` attribute that returns a 2-tuple: The version number
+(``0``) and the address of ``cudaStream_t``:
```
Comment (shwina) [inline cuda_core/docs/source/interoperability.rst]: We should clarify here that the address is a Python `int`.

**Review by shwina — DISMISSED** [review]
Approving pending some small suggestions.

File: cuda_core/docs/source/interoperability.rst (hunk 1 above)
Comment (leofang) [reply cuda_core/docs/source/interoperability.rst]: I removed the mention of `cudart` and added a hyperlink to the cudart docs in commit c06270c92b4a779314a029ac75c1645ab242b00c.

File: cuda_core/docs/source/interoperability.rst (hunk 2 above)
Comment (leofang) [reply cuda_core/docs/source/interoperability.rst]: Noted that both are Python int in c06270c92b4a779314a029ac75c1645ab242b00c.

**Comment by leofang** [comment]
@shwina merge?

**Review by vzhurba01 — APPROVED** [review]
LGRM

**Comment by leofang** [comment]
Thanks guys!

**Pull request merged**